internal/app/storage/fileHandler.go

```diff
@@ -0,0 +1,65 @@
+package storage
+
+import (
+	"encoding/json"
+	"os"
+)
+
+type Record struct {
+	Hash string `json:"hash"`
+	URL  string `json:"url"`
+}
+
+type producer struct {
+	file    *os.File
+	encoder *json.Encoder
+}
+
+func NewProducer(filename string) (*producer, error) {
+	file, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0777)
+	if err != nil {
+		return nil, err
+	}
+
+	return &producer{
+		file:    file,
+		encoder: json.NewEncoder(file),
+	}, nil
+}
+
+func (p *producer) WriteRecord(record *Record) error {
+	return p.encoder.Encode(&record)
```

Comment:
Лично я не уверен, что encoder защищен от DATA RACE
Что будет, если 10К горутин будут дергать WriteRecord?

Reply:
речь опять про многопоточность. Вроде ж ее еще не проходили?

Reply:
пишут что посмотрели в код и не нашли общих переменных
https://stackoverflow.com/questions/50257026/are-jsondecoder-and-jsonencoder-classes-thread-safe

Judging by the code for JSONDecoder and JSONEncoder they seem to be thread-safe.
Both encode() and decode() use JSONSerialization (which has been thread-safe since iOS 7 and macOS 10.9), and both methods create their own private decode/encode objects in local variables. There is no shared storage as far as I can see.

Reply:
под многопоточностью обычно подразумевают параллельность и конкурентность 
ну да, много клиентов ломятся в мой сервис ... вроде это и рождает (может родить) многопоточность, которая приводит к проблемам конкурентного доступа.